devito/equation.py

```diff
@@ -60,12 +61,10 @@ def __new__(cls, *args, **kwargs):
         obj = sympy.Eq.__new__(cls, *args, **kwargs)
         obj._subdomain = subdomain
         obj._substitutions = substitutions
-        functions = retrieve_functions(obj)
-        functions = [f for f in functions if not f.is_SparseFunction]
-        if any(f.coefficients == 'symbolic' for f in functions):
+        if obj._symbolic_coefficients:
```

Comment:
I **really** like this change!

Reply:
I think there's a point in ``finite_difference.py`` where we can do the same?

Reply:
Yes, in the `check_symbolic` decorator. Thanks for reminding me, will update.